src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/logs/logs.component.ts

```diff
@@ -59,6 +64,24 @@ export class LogsComponent implements OnInit, OnDestroy {
     clearInterval(this.interval);
   }
 
+  getDaemonDetails() {
+    this.cephService.getDaemons('loki').subscribe((data: any) => {
+      if (data.length > 0) {
+        this.lokiServiceStatus = data[0].status;
+      } else {
+        this.lokiServiceStatus = 0;
+      }
+    });
+
+    this.cephService.getDaemons('promtail').subscribe((data: any) => {
+      if (data.length > 0) {
+        this.promtailServiceStatus = data[0].status;
+      } else {
+        this.promtailServiceStatus = 0;
+      }
+    });
```

Comment:
I'm always suggesting that you guys rely as much as possible in the AsyncPipe instead of managing the subscriptions yourselves (one often forgets to unsubscribe from you observables, and while it's not always strictly necessary, once one gets used not to do it, you also forget it when it's mandatory):
```suggestion
    this.lokiServiceStatus$ = this.cephService.getDaemons('loki');
    this.promtailServiceStatus$ = this.cephService.getDaemons('promtail');
```
And in the [HTML template](https://stackoverflow.com/questions/47391735/ngif-with-multiple-async-pipe-variables):
```html
<ng-container *ngIf="lokiServiceStatus$ | async as lokiServiceStatus; else daemonLogsTpl">
  <div *ngIf="promtailServiceStatus$ | async as promtailServiceStatus; else daemonLogsTpl">
```
For this to work you must ensure that the async call fails if the status is not ok (instead of returning an empty status, which btw is a weird way of communicating: 'if you ask me if I'm ok: if I tell you "Yes", then I am; but I tell you nothing, then "I'm not"...'.

Without modifying the service you can achieve the same behaviour but adding the logic to `.pipe(map())`, instead of to the `.subscribe()`.